dbt/adapters/snowflake/relation_configs/dynamic_table.py

```diff
@@ -55,11 +60,12 @@ class SnowflakeDynamicTableConfig(SnowflakeRelationConfigBase):
     query: str
     target_lag: str
     snowflake_warehouse: str
+    catalog: SnowflakeCatalogConfig
     refresh_mode: Optional[RefreshMode] = RefreshMode.default()
     initialize: Optional[Initialize] = Initialize.default()
 
     @classmethod
-    def from_dict(cls, config_dict) -> "SnowflakeDynamicTableConfig":
+    def from_dict(cls, config_dict: Dict[str, Any]) -> Self:
```

Comment:
I tried adding Self but it said this wasn't allowed for our current python version!? Maybe because you're using `typing_extensions` instead of `typing`?

Reply:
> Maybe because you're using typing_extensions instead of typing

Correct, `Self` isn't in `typing` until something like 3.11.